docs/tutorial/where-to-store-app-data.md

```diff
@@ -0,0 +1,93 @@
+# Where to Store App Data
+
+This tutorial provides guidance on where to store user data properly depending
+on the app's type.
+
+## User data paths
+
+In Electron, locations of user data can be obtained and customized with
+`app.getPath` and `app.setPath` APIs, and there are 3 types of paths are used
+most commonly: `userData`, `localUserData` and `userCache`.
+
+### `userData`
+
+The `userData` path is where app should store persistent user data, such as
+config files and authentication information. Generally you should not write
+large files in this location as some environments may choose to synchronize
+files in this location to cloud, especially on Window.
+
+It by default points to:
+
+* `%APPDATA%/AppName` on Windows
+* `$XDG_CONFIG_HOME/AppName` or `~/.config/AppName` on Linux
+* `~/Library/Application Support/AppName` on macOS
+
+### `localUserData`
+
+On Linux and macOS `localUserData` is an alias to `userData`, on Windows it
+points to `%LOCALAPPDATA%/AppName` instead.
+
+On Windows, both `localUserData` and `userData` can be used to store persistent
+user data, the difference between them is, the `userData` can be synchronized to
+cloud, but `localUserData` can not.
+
+The convention is, if the user data takes lots of disk space, they should be
+written to `localUserData`, otherwise it can be written to `userData` instead.
+
+### `userCache`
+
+The `userCache` path is where cache data should be stored, files in this
+location should be safe to be removed without affecting app's state.
+
+On Linux it points to `$XDG_CACHE_HOME/AppName` or `~/.cache/AppName`.
+
+On macOS it points to `~/Library/Caches/AppName`.
+
+On Windows there is no specialized location to store cache data, and most apps
+write cache data to `%LOCALAPPDATA%`, so this path points to `localUserData` by
+default. If your app is writing files to both `localUserData` and `userCache`,
+you should consider changing the `userCache` path to a subdirectory of
+`localUserData` to avoid conflicts.
+
+## Data in Electron apps
+
+Data generated by a typical Electron app can be categorized to 3 kinds:
+
+1. Browsing data created by `Session` API, which in most apps are generated
+   implicitly when using the `BrowserWindow` API. They can be roughly separated
+   in 2 kinds:
+   1. HTTP disk cache and JavaScript code cache, which are written into the
+      `userCache` directory.
+   2. All other kinds of browsing data, such as cookies, localStorage, indexDB,
+      GPU Cache, and dictionaries. They are all considered as browsing data in
+      Chromium and written into the `sessionData` directory, which points to
+      `userData` by default.
+2. Data implicitly written by other Electron APIs, such as crash reports written
```

Comment:
```suggestion
1. Data implicitly written by other Electron APIs, such as crash reports written
```

nit: the MD spec handles numbering automatically, so it's easier to just number everything with `1.` and let the parser do the numbering work (makes the diff smaller if you want to insert items in a refactor for example)